### PR TITLE
provisioning of swarm and k8s on aws, azure, google, exoscale

### PR DESCRIPTION
**Pull request by konstan**
Please review but hold on with merging if Nuvla release is planned in a couple of days. I'm validating this on staging against our prod DB to see what migration is required.

### CODE REVIEW

File: code/src/sixsq/nuvla/server/resources/credential_infrastructure_service_kubernetes.clj [hunk 1]
```diff
@@ -0,0 +1,89 @@
+(ns sixsq.nuvla.server.resources.credential-infrastructure-service-kubernetes
```
Comment (0xbase12) [inline code/src/sixsq/nuvla/server/resources/credential_infrastructure_service_kubernetes.clj:1]: exactly the same as infrastructure-service-swarm. We have to choose if we keep both or rename. We have to  after decision, update existing creds if needed and adapt UI accordingly.

File: code/src/sixsq/nuvla/server/resources/infrastructure_service_coe.clj [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+;; post-add hook that creates a job that will deploy Container Orchestration Engine
+;; corresponding to :subtype of the infrastructure service.
+;;
+
+(defmethod infra-service/post-add-hook method
+  [service request]
+  (try
+    (when-not (contains? service :management-credential)
+      (throw (r/ex-response (format ":management-credential required to create COE %" (:subtype service))
+                            500 (:id service))))
```
Comment (0xbase12) [inline code/src/sixsq/nuvla/server/resources/infrastructure_service_coe.clj:72]: should it be a 400 instead

Comment (konstan) [reply code/src/sixsq/nuvla/server/resources/infrastructure_service_coe.clj:72]: I suggest 412 Precondition Failed.

File: code/src/sixsq/nuvla/server/resources/credential_infrastructure_service_kubernetes.clj (hunk 1 above)
Comment (konstan) [reply code/src/sixsq/nuvla/server/resources/credential_infrastructure_service_kubernetes.clj:1]: Yes, they are the same. I didn't want to address this during the development of the feature. Let's do this as a next step.

File: code/src/sixsq/nuvla/server/resources/infrastructure_service_coe.clj (hunk 2 above)
Comment (konstan) [reply code/src/sixsq/nuvla/server/resources/infrastructure_service_coe.clj]: Chose to return 412. Done in c433fca.

File: code/src/sixsq/nuvla/server/resources/credential_infrastructure_service_kubernetes.clj (hunk 1 above)
Comment (konstan) [reply code/src/sixsq/nuvla/server/resources/credential_infrastructure_service_kubernetes.clj:1]: Created this issue as followup https://github.com/nuvla/api-server/issues/372

**Pull request merged**